src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -201,21 +215,74 @@ def process(
                     cc.message(grant_statements)
                     collected_output.append(grant_statements)
 
-                self.deannotate(py_file, extension_fns)
+                with open(sql_file, "a") as file:
```

Comment:
why append? Isn't it a new file?

Reply:
It shouldn't be possible. If the user already has a generated dir, we should refuse to generate _anything_ (in the compiler) unless they change the generated dir.

Reply:
I am not sure if this refusal should be at the compiler level just yet, given we don't know how other processors may be written in the future or what their behavior might be.
So I have added the refusal just to the Snowpark Processor for now, before we even start doing any of the collection work. Idea being, if you're not writing the collected functions into a file, then why are you collecting the functions at all? Printing to console is one reason, but not one I'm convinced needs all this work.

Reply:
Fair enough